crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -0,0 +1,331 @@
+            sender: tx,
+            #[cfg(feature = "read-tx-timeouts")]
+            read_transactions: None,
+        };
+
+        txn_manager.start_message_listener(env, rx);
+
+        txn_manager
+    }
+
+    fn start_message_listener(&self, env: EnvPtr, rx: Receiver<TxnManagerMessage>) {
+        std::thread::spawn(move || {
+            #[allow(clippy::redundant_locals)]
+            let env = env;
+            loop {
+                match rx.recv() {
+                    Ok(msg) => match msg {
+                        TxnManagerMessage::Begin { parent, flags, sender } => {
+                            let mut txn: *mut ffi::MDBX_txn = ptr::null_mut();
+                            sender
+                                .send(
+                                    mdbx_result(unsafe {
+                                        ffi::mdbx_txn_begin_ex(
+                                            env.0,
+                                            parent.0,
+                                            flags,
+                                            &mut txn,
+                                            ptr::null_mut(),
+                                        )
+                                    })
+                                    .map(|_| TxnPtr(txn)),
+                                )
+                                .unwrap();
+                        }
+                        TxnManagerMessage::Abort { tx, sender } => {
+                            let result = mdbx_result(unsafe { ffi::mdbx_txn_abort(tx.0) });
+                            sender.send(result).unwrap();
+                        }
+                        TxnManagerMessage::Commit { tx, sender } => {
+                            sender
+                                .send({
+                                    let mut latency = CommitLatency::new();
+                                    mdbx_result(unsafe {
+                                        ffi::mdbx_txn_commit_ex(tx.0, latency.mdb_commit_latency())
+                                    })
+                                    .map(|v| (v, latency))
+                                })
+                                .unwrap();
+                        }
+                    },
+                    Err(_) => return,
+                }
+            }
+        });
+    }
+
+    pub(crate) fn send_message(&self, message: TxnManagerMessage) {
+        self.sender.send(message).unwrap()
+    }
+}
+
+#[cfg(feature = "read-tx-timeouts")]
+mod read_transactions {
+    use crate::{error::mdbx_result, txn_manager::TxnManager, Error};
+    use dashmap::{DashMap, DashSet};
+    use std::{
+        sync::Arc,
+        time::{Duration, Instant},
+    };
+    use tracing::{error, trace, warn};
+
+    const READ_TRANSACTIONS_CHECK_INTERVAL: Duration = Duration::from_millis(500);
+
+    impl TxnManager {
+        pub(crate) fn with_max_read_transaction_duration(
+            mut self,
+            duration: Duration,
```

Comment:
there should be a reference to the DEFAULT_MAX_READ_TRANSACTION_DURATION somewhere

Reply:
any ideas where to mention it, so when you build the environment, it's obvious that unless a custom `max_read_transaction_duration` is specified, `DEFAULT_MAX_READ_TRANSACTION_DURATION` will be used?

Reply:
I want this ideally on the field as a comment, so I see what the default is.
otherwise, I need to check the callgraph and find the default like that